cfn-resources/project-ip-access-list/cmd/resource/create.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+)
+
+func CreateOp(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	if errEvent := validateModel(CreateRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	if len(currentModel.AccessList) == 0 {
+		return progressevents.GetFailedEventByCode("AccessList must not be empty", cloudformation.HandlerErrorCodeInvalidRequest), nil
+	}
+
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	// Create atlas client
+	client, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	event, err := createEntries(currentModel, client)
+	if event.OperationStatus == handler.Failed || err != nil {
+		return event, nil
+	}
+
+	_, _ = logger.Debugf("Create --- currentModel:%+v", currentModel)
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create Complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func createEntries(model *Model, client *util.MongoDBClient) (handler.ProgressEvent, error) {
+	request, err := newPaginatedNetworkAccess(model)
+	if err != nil {
+		return handler.ProgressEvent{
+			Message:          "Error in parsing the resource schema",
+			OperationStatus:  handler.Failed,
+			HandlerErrorCode: cloudformation.HandlerErrorCodeAlreadyExists}, err
+	}
+
+	projectID := *model.ProjectId
+
+	if isEntryAlreadyInAccessList, err := isEntryAlreadyInAccessList(client, model); isEntryAlreadyInAccessList || err != nil {
```

Comment:
do we need this call or the second one will fail with "existing resource" error?

Reply:
yes, we need it because the POST endpoint does not fail if we try to create an entry that is already there but it return a 200 response